packages/orchestrator-ui-components/src/components/WfoSubscription/WfoSubscriptionGeneral.tsx

```diff
@@ -24,71 +25,72 @@ export const WfoSubscriptionGeneral = ({
-            },
-            {
-                key: t('customerUuid'),
-                value:
-                    subscriptionDetail && subscriptionDetail.customer
-                        ? `${subscriptionDetail.customer?.customerId}`
-                        : '-',
-                textToCopy: subscriptionDetail.customer?.customerId,
-            },
-            {
-                key: t('note'),
-                value: (subscriptionDetail && subscriptionDetail.note) || '-',
-            },
-        ];
-    };
+    const getSubscriptionDetailBlockData = (): WfoKeyValueTableDataType[] => [
+        {
+            key: t('subscriptionId'),
+            value: subscriptionDetail.subscriptionId,
+            textToCopy: subscriptionDetail.subscriptionId,
+        },
+        {
+            key: t('productName'),
+            value: subscriptionDetail.product.name,
+        },
+        {
+            key: t('description'),
+            value: subscriptionDetail.description,
+        },
+        {
+            key: t('startDate'),
+            value: formatDate(subscriptionDetail.startDate),
+        },
+        {
+            key: t('endDate'),
+            value: formatDate(subscriptionDetail.endDate),
+        },
+        {
+            key: t('status'),
+            value: (
+                <WfoSubscriptionStatusBadge
+                    status={subscriptionDetail.status}
+                />
+            ),
+        },
+        {
+            key: t('insync'),
+            value: <WfoInSyncField subscriptionDetail={subscriptionDetail} />,
+        },
+        {
+            key: t('customer'),
+            value:
+                subscriptionDetail && subscriptionDetail.customer
+                    ? `${subscriptionDetail.customer?.fullname}`
+                    : '-',
+        },
+        {
+            key: t('customerUuid'),
+            value:
+                subscriptionDetail && subscriptionDetail.customer
+                    ? `${subscriptionDetail.customer?.customerId}`
+                    : '-',
+            textToCopy: subscriptionDetail.customer?.customerId,
+        },
+        {
+            key: t('note'),
+            value: (subscriptionDetail && subscriptionDetail.note) || '-',
+        },
+    ];
+
+    const getMetadataBlockData = (): WfoKeyValueTableDataType[] =>
+        Object.entries(subscriptionDetail.metadata).map(([key, value]) => ({
+            key: camelToHuman(key),
+            value: <WfoJsonCodeBlock data={value} isBasicStyle />,
+        }));
 
-    const getFixedInputBlockData = (): WfoKeyValueTableDataType[] => {
-        return subscriptionDetail.fixedInputs.map((fixedInput) => ({
+    const getFixedInputBlockData = (): WfoKeyValueTableDataType[] =>
+        subscriptionDetail.fixedInputs.map((fixedInput) => ({
             key: fixedInput.field,
             value: fixedInput.value,
         }));
-    };
 
     const getProductInfoBlockData = (): WfoKeyValueTableDataType[] => {
```

Comment:
Maybe rename this to subscriptionInfo instead of productInfo

Reply:
Its information about the product though right?